x-pack/plugins/enterprise_search/server/lib/check_access.test.ts

```diff
@@ -29,24 +42,36 @@ describe('checkAccess', () => {
       },
     },
   };
+  const mockSpaces = spacesMock.createSetup();
   const mockDependencies = {
-    request: {},
+    request: { auth: { isAuthenticated: true } },
     config: { host: 'http://localhost:3002' },
     security: mockSecurity,
+    spaces: mockSpaces,
   } as any;
 
   describe('when security is disabled', () => {
-    it('should allow all access', async () => {
+    it('should allow all access when enabled at the space', async () => {
```

Comment:
I'd like to clean up the tests if possible (+ I noticed a few uncovered lines). Based on my understanding of the new branching path, I think the test file should now be organized into 3 main top-level describe blocks:

  - Space disabled
  - Spaces plugin unavailable (or error-ing)
  - Space enabled (most previous tests go under this new block)

I didn't want to put that on you though so I spiked out the changes I wanted:

https://github.com/constancecchen/kibana/commit/f1b3f2aff733cb84508708d4412cdc83a9f5dba8

Feel free to cherry-pick that into this PR, or if you strongly prefer I can open a separate follow-up PR after this one gets merged in.

Reply:
@constancecchen thank you so much for taking the time to put a commit together for your requested changes! Not at all necessary, but very much appreciated ❤️